src/server/pkg/tarutil/util.go

```diff
@@ -0,0 +1,251 @@
+package tarutil
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tar"
+)
+
+type File interface {
+	Header() (*tar.Header, error)
+	Content(io.Writer) error
+}
+
+type file struct {
+	hdr  *tar.Header
+	data []byte
+}
+
+func NewFile(name string, data []byte) File {
```

Comment:
Maybe call this `NewMemFile` or `NewInMemoryFile`.  This shouldn't be used for large files.

Reply:
I'll go with `NewMemFile`.